### PR TITLE
V1.3.0 status api

### PR DESCRIPTION
**Pull request by ryandoherty**
This PR adds a status screen that shows the full status of the connected RCP. It polls RCP every 2 seconds to always have the latest status (due to things like GPS and cellular changing).

Todo: 
- ~~Figure out what icon to use in the side menu (and how to get the correct code to use it)~~
- I'd like to build a better 'Loading...' screen, but very minor nitpick
- Tests!

### CODE REVIEW

File: autosportlabs/racecapture/views/status/statusview.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+    def __init__(self, track_manager, rc_api, **kwargs):
+        super(StatusView, self).__init__(**kwargs)
+        self.track_manager = track_manager
+        self.rc_api = rc_api
+        self.register_event_type('on_tracks_updated')
+        self.rc_api.addListener('status', self.on_status_updated)
+        self.menu_node = self.ids.menu
+        self.menu_node.bind(selected_node=self._on_menu_select)
+
```
Comment (brentpicasso) [inline autosportlabs/racecapture/views/status/statusview.py]: should prefix any of these with underscores if they're  meant as private only to class

File: autosportlabs/racecapture/views/status/statusview.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+    selected_item = None
+
+    menu_built = False
+
+    #Track manager for getting track name
+    track_manager = None
+
+    #Used for building the left side menu
+    menu_keys = {
+        "system": "System Status",
```
Comment (brentpicasso) [inline autosportlabs/racecapture/views/status/statusview.py]: suggest _menu_keys here as well unless it's accessed outside of this class

File: autosportlabs/racecapture/views/status/statusview.py [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+        "cell": "Cellular",
+        "bt": "Bluetooth",
+        "logging": "Logging",
+        "track": "Track",
+        "telemetry": "Telemetry"
+    }
+
+    #Dict for getting English text for status enums
+    enum_keys = {
+        'GPS': {
```
Comment (brentpicasso) [inline autosportlabs/racecapture/views/status/statusview.py]: ditto

File: autosportlabs/racecapture/views/status/statusview.py (hunk 1 above)
Comment (ryandoherty) [inline autosportlabs/racecapture/views/status/statusview.py]: :+1:

File: autosportlabs/comms/comms.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
 COMMAND_KEEP_ALIVE = 'PING'
 
 def connection_process_message_reader(rx_queue, connection, should_run):
-    print('connection process message reader started')
+    #print('connection process message reader started')
```
Comment (brentpicasso) [inline autosportlabs/comms/comms.py]: Was the commenting out of the logging info intentional or did it slip in a commit?

If you have time, might recommend adding in the Kivy Logging infrastructure as we've done here already: https://github.com/autosportlabs/RaceCapture_App/blob/master/service/main.py#L36

Comment (ryandoherty) [inline autosportlabs/comms/comms.py]: I did comment it out during development b/c there's so much logging and it makes it hard to debug. I can add it back in if needed. 

I agree, I want to use a 'real' logging tool so we can handle it better.

Comment (brentpicasso) [inline autosportlabs/comms/comms.py]: The kivy logging is super easy to implement, can almost be a search / replace.  only thing they want you to do is prefix it with the "area" being logged. e.g.  Logger.info('status_view:  or  Logger.info('comms:

Comment (ryandoherty) [inline autosportlabs/comms/comms.py]: like

Comment (ryandoherty) [inline autosportlabs/comms/comms.py]: Will split out logging stuff into a separate PR.
